tests/network/testdata/vlans_and_bonds.txt

```diff
@@ -0,0 +1,404 @@
+[!exec:bash] stop
+[!exec:grep] stop
+[!exec:cut] stop
+[!exec:curl] stop
+[!exec:sleep] stop
+
+{{$srvimage := "docker://lfedge/eden-docker-test:83cfe07"}}
+{{$appimage := "docker://lfedge/eden-eclient:d9eb23f"}}
+{{define "ssh"}}ssh -o ConnectTimeout=10 -o StrictHostKeyChecking=no -o PasswordAuthentication=no -i {{EdenConfig "eden.tests"}}/eclient/image/cert/id_rsa root@{{end}}
+
+# In this test we first create two ethernet loops (pairs of eth interfaces interconnected
+# using virtual crossover cables). On both ends of these cables we put bonds to aggregate
+# them together as one link (with RR policy). On one end we then configure two VLAN sub-interfaces
+# to be used as EVE uplinks (VLAN 100 & 200), while the other end is attached to a switch network
+# with applications separated by these VLANs and with a DHCP server connected by a trunk port,
+# providing IP allocation separately for each VLAN.
+# If bonds and VLAN sub-interfaces work correctly, then applications as well as uplink ports should
+# receive IP addresses from the DHCP server and it should be possible to ping between applications
+# and EVE through these VLANs.
+#
+# Network topology:
+#
+#  bond1.100 (EVE uplink) --+           +-- eth2 -- (eth loop) -- eth3 --+                        +-- (trunk port) -- DHCP server
+#                           |           |                                |                        |                   (with VLAN sub-interfaces inside)
+#                           +-- bond1 --+                                +-- bond2 -- switch NI --+-- (VLAN 100) -- app1
+#                           |           |                                |                        |
+#  bond1.200 (EVE uplink) --+           +-- eth4 -- (eth loop) -- eth5 --+                        +-- (VLAN 200) -- app2
+#
+#
+
+
+# The application providing DHCP services for VLANs requires netadmin capabilities, otherwise
+# it is not permitted to create VLAN sub-interfaces. For security reasons, EVE does not allow
+# to grant these capabilities to native containers, therefore it is required to deploy apps
+# as VMs-in-Containers for this test to pass.
+exec -t 2m bash check_vm_support.sh
+source .env
+[!env:with_hw_virt] skip 'Missing HW-assisted virtualization capability'
+
+# This test is based on ethernet loops, which are only supported on the QEMU dev model.
+{{$devmodel := EdenConfig "eve.devmodel"}}
+{{if not (eq $devmodel "ZedVirtual-4G")}}
+skip 'Test is supported only on QEMU'
+{{end}}
```

Comment:
Unfortunately right now this combination (qemu+HW-assisted virtualization) is not handled by any CI workflows. Can we not use dhcp server in that test to make it a part of non-HW-accelerated (per PR) workflow?

Reply:
Unfortunately without Netdev capabilities inside applications I cannot think of any proper test for more advanced network features, such as VLAN and LAGs. And eventually we will have ROL where we will be able to run these tests. Testing applications as VMs inside containers (and not just containers) as part of CI should be our goal in any case.